core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -1953,6 +2028,15 @@ private RecordCursorFactory generateTableQuery(
         }
     }
 
+    private boolean requiresTimestamp(int joinType) {
```

Comment:
I mean sparse array. I wouldn't think loop is faster than || comparison. Probably not. Using sparse array lookup will look like:

```java
boolean requiresTimeatamp = joinsRequiringTimestamp[model.getJoinType()];
```